### PR TITLE
Fix Unity app hooking by preventing LibRevive from being freed

### PR DESCRIPTION
**Pull request by rjoudrey**

### CODE REVIEW

File: REV_CAPI.cpp [hunk 1]
```diff
@@ -23,12 +23,14 @@ OVR_PUBLIC_FUNCTION(ovrResult) ovr_Initialize(const ovrInitParams* params)
...
 	MH_QueueDisableHook(LoadLibraryW);
 	MH_QueueDisableHook(OpenEventW);
+	MH_QueueDisableHook(FreeLibrary);
 	MH_ApplyQueued();
 
 	g_VRSystem = vr::VR_Init(&g_InitError, vr::VRApplication_Scene);
 
 	MH_QueueEnableHook(LoadLibraryW);
 	MH_QueueEnableHook(OpenEventW);
+	MH_QueueEnableHook(FreeLibrary);
```
Comment (LibreVR) [inline REV_CAPI.cpp]: It's good to see you remembered it, but it's not actually necessary to disable this hook.

File: main.cpp [hunk 2]
```diff
@@ -44,6 +46,23 @@ HANDLE WINAPI HookOpenEvent(DWORD dwDesiredAccess, BOOL bInheritHandle, LPCWSTR
...
+	GetModuleFileName(hModule, lpFileName, sizeof(lpFileName));
+	LPCWSTR fileName = PathFindFileNameW(lpFileName);
+
+	WCHAR lpReviveFileName[MAX_PATH];
+	GetModuleFileName((HMODULE)ReviveModule, lpReviveFileName, sizeof(lpReviveFileName));
+	LPCWSTR reviveFileName = PathFindFileNameW(lpReviveFileName);
+
+	if (wcscmp(fileName, reviveFileName) == 0) {
+		return true;
+	}
```
Comment (LibreVR) [inline main.cpp]: You don't need to do all this, this function will be called with the module handle we returned in `HookLoadLibrary`. So you can simply do this here:

```
if (hModule == ReviveModule)
    return true;
```

**Comment by LibreVR** [comment]
If you can make those two changes, then it's ready to be merged.

**Comment by rjoudrey** [comment]
Yeah, I should've read the MH_QueueEnableHook documentation before blindly putting it there.

**Comment by rjoudrey** [comment]
Just noticed that there is still some unused code there. Fixing it now.

**Comment by LibreVR** [comment]
@rjoudrey Just tested it, it works fine with Lucky's Tale. Nice!

**Pull request merged**